odoo/addons/base/models/ir_ui_view.py

```diff
@@ -370,6 +371,17 @@ def _search_model_data_id(self, operator, value):
         data = self.env['ir.model.data'].sudo().search(domain)
         return [('id', 'in', data.mapped('res_id'))]
 
+    @api.depends('model')
+    def _compute_model_id(self):
+        result = self.env['ir.model'].sudo().search([('model', 'in', self.mapped('model'))])
+        result = {r.model: r.id for r in result}
```

Comment:
Nowadays I try to avoid having a variable changing type.
```suggestion
        result = {
            m.model: m.id
            for m in self.env['ir.model'].sudo().search([('model', 'in', self.mapped('model'))])
        }
```